examples/multi_processor/main.cc

```diff
@@ -0,0 +1,79 @@
+#include "opentelemetry/sdk/trace/simple_processor.h"
+#include "opentelemetry/sdk/trace/tracer_context.h"
+#include "opentelemetry/sdk/trace/tracer_provider.h"
+#include "opentelemetry/trace/provider.h"
+
+// Using an exporter that simply dumps span data to stdout.
+#include "foo_library/foo_library.h"
+#include "opentelemetry/exporters/memory/in_memory_span_exporter.h"
+#include "opentelemetry/exporters/ostream/span_exporter.h"
+
+using opentelemetry::exporter::memory::InMemorySpanExporter;
+
+InMemorySpanExporter *memory_span_exporter;
+
+namespace
+{
+void initTracer()
+{
+  auto exporter1 = std::unique_ptr<sdktrace::SpanExporter>(
+      new opentelemetry::exporter::trace::OStreamSpanExporter);
+  auto processor1 = std::unique_ptr<sdktrace::SpanProcessor>(
+      new sdktrace::SimpleSpanProcessor(std::move(exporter1)));
+
+  auto exporter2 = std::unique_ptr<sdktrace::SpanExporter>(new InMemorySpanExporter());
+
+  // fetch the exporter for dumping data later
+  memory_span_exporter = dynamic_cast<InMemorySpanExporter *>(exporter2.get());
```

Comment:
Assign `new InMemorySpanExporter()` to memory_span_exporter` directory before creating unique_ptr, so avoid this `dynamic_cast`?

Reply:
Good point. Yeah, it can be done, but I felt it's more important to have `exporter2` created the way the application developer would create, as `exporter2` needs to be added to `TracerProvider` afterward.  The statement:

`auto exporter2 = std::unique_ptr<sdktrace::SpanExporter>(memory_span_exporter );`

would be confusing for application devs as they would never create an exporter this way. Other than that, I don't have any issue in changing the code as suggested :)